comm/p2p/libp2p.go

```diff
@@ -69,23 +69,33 @@ func (c Libp2pCommunication) Broadcast(
 	marshaledMsg, err := json.Marshal(wMsg)
 	if err != nil {
 		c.logger.Error().Err(err).Str("SessionID", sessionID).Msg("unable to marshal message")
-		return
+		return err
 	}
 	c.logger.Debug().Str("MsgType", msgType.String()).Str("SessionID", sessionID).Msg(
 		"broadcasting message",
 	)
+
+	p := pool.New().WithErrors().WithFirstError()
```

Comment:
This would kill other broadcast attempts as soon as one attempt fails, which is not expected behavior.

Reply:
This only returns the first error and doesn't cancel.